server/src/main/java/org/elasticsearch/common/settings/Settings.java

```diff
@@ -876,6 +876,30 @@ public Set<String> keySet() {
         return newKeySet;
     }
 
+    /*
+     * This method merges the given newSettings into this Settings, returning either a new Settings object or
+     * this if the newSettings are null or empty. If any values are null in newSettings, those keys are removed
+     * from the returned object.
+     */
+    public Settings merge(Settings newSettings) {
+        Objects.requireNonNull(newSettings);
+        if (Settings.EMPTY.equals(newSettings)) {
+            return this;
+        }
+        Settings.Builder builder = Settings.builder().put(this).put(newSettings);
```

Comment:
the put(newSettings) can be removed since individual settings are added below, right?

Reply:
Oops, yes. That was a mistake!